src/utils/agent_tools.py

```
@@ -0,0 +1,1710 @@
+                limit=10,
+                context_window=0,  # Just the matching message
+            )
+            for matches, _ in snippets:
+                for msg in matches:
+                    if msg.peer_name == observed:
+                        content_key = msg.content[:100].lower()
+                        if content_key not in seen_content:
+                            seen_content.add(content_key)
+                            # Extract the instruction
+                            instructions.append(f"'{msg.content.strip()}'")
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)